exercises/practice/largest-series-product/.meta/example.jl

```diff
@@ -0,0 +1,16 @@
+function largest_product(s::AbstractString, span::Int)
+    0 <= span <= length(s) || throw(DomainError(span, "Span must be ≥ 0 and ≤ length(s)"))
+    (iszero(span) || isempty(s)) && return 1
+
+    digits = zeros(Int8, span)
+    largest = 0
+
+    for c in s
+        isdigit(c) || throw(DomainError(s, "String must only contain digits"))
+        popfirst!(digits)
+        push!(digits, c - '0')
+        largest = max(largest, prod(digits))
+    end
+
+    largest
+end
```

Comment:
I'm new to Julia so let me know if this is bad

Reply:
I don't see anything critically wrong with it.

You could move the `isdigit` check up in the style of a guard clause (`all(isdigit, collect(s)) || throw(...)`). It's slower but might be cleaner.

Reply:
There's nothing especially wrong with it, though I think most people would use a series of views rather than pushing and popping from a vector.

I wrote an approaches doc for this exercise and you can see a few variations on how I solved the exercise there: https://github.com/exercism/julia/pull/564/files#diff-1e0a7d63ad5259cf1372acd0aedb41baae1d5ebc78add316c25473f3d4995b33